runtime/gc_structs/ContinuationSlotIterator.cpp

```diff
@@ -0,0 +1,57 @@
+
+/*******************************************************************************
+ * Copyright IBM Corp. and others 2025
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] https://openjdk.org/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
+ *******************************************************************************/
+
+/**
+ * @file
+ * @ingroup GC_Structs
+ */
+
+#include "j9.h"
+#include "j9cfg.h"
+
+#include "ContinuationSlotIterator.hpp"
+
+/**
+ * @return the next slot in the J9VMContinuation
+ * @return NULL if there are no more such slots
+ */
+j9object_t *
+GC_ContinuationSlotIterator::nextSlot()
+{
+	j9object_t *ret = NULL;
+#if JAVA_SPEC_VERSION >= 24
+	if (_monitorRecord != NULL) {
+		J9MonitorEnterRecord *currentMonitorRecord = _monitorRecord;
+		_monitorRecord = currentMonitorRecord->next;
+		ret = &currentMonitorRecord->object;
+	} else if (_jniMonitorRecord != NULL) {
+		J9MonitorEnterRecord *currentMonitorRecord = _jniMonitorRecord;
+		_jniMonitorRecord = currentMonitorRecord->next;
+		ret =  &currentMonitorRecord->object;
+	} else if (NULL != _vthread) {
+		ret = _vthread;
+		_vthread = NULL;
```

Comment:
i'll defer this to @dmitripivkine if this chained if is good enough (only 3 cases) or we want some kind of state or index to drive through 3 cases more directly (like in GC_VMThreadIterator::nextSlot() )

Reply:
I think this implementation is clear enough. We can create iterator states if you like.

Reply:
ok, let's keep it 
but some formatting should be cleaned up:
- NULL should be left of binary operators
- there is extra empty spaces at line 50

Reply:
now that you have a state, this chained-if can turn into a switch

Reply:
hm, I just realized that you are sharing the state for 2 cases (not sure why)
so even if you have switch it would be only 2 states and monitor state would have and if inside, so not much different than 3 chained ifs
you can leave it as-is if you still like it